### PR TITLE
Support network scoped service aliases

### PR DESCRIPTION
**Pull request by lisongmin**
To support network scoped aliases, we using the `alias` option in `--network`.
Since the `--network-alias` will override the `alias` set by `--network`,
we do not using the `--network-alias` any more. All aliases will be set
via `--network`.

### CODE REVIEW

File: tests/integration/test_podman_compose_network_scoped_aliases.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
+# SPDX-License-Identifier: GPL-2.0
```
Comment (p12tic) [inline tests/integration/test_podman_compose_network_scoped_aliases.py:1]: It is fine to not have integration test, if the unit tests cover the code like in this case. Thanks for adding it anyway.

**Comment by p12tic** [comment]
@lisongmin Thanks for the PR. I will write several comments, but overall it looks good.

Please note, that such a PR where all changes are added as a single commit is extremely hard to review. Refactoring is mixed with functionality changes and as a result I would have to replay everything in my head.

I've split the PR in small commits that refactor one thing at a time. The end result is almost the same except several [deliberate changes](https://github.com/containers/podman-compose/compare/873c1c0f18cb59402a466001bdb39769233a95d5..681d636eb70a93294be97b39ebd462bdd5fdb742). The functionality changes are then way smaller and way easier to review.

File: podman_compose.py [hunk 2]
```diff
@@ -855,8 +855,9 @@ async def assert_cnt_nets(compose, cnt):
     create missing networks
     """
     net = cnt.get("network_mode")
-    if net and not net.startswith("bridge"):
+    if net:
```
Comment (p12tic) [inline podman_compose.py:858]: Could you please tell me why is this change needed. It's not obvious to me. If there's some requirement in compose spec that I'm not aware of, please point me to it.

File: podman_compose.py [hunk 3]
```diff
@@ -910,12 +911,9 @@ def get_net_args_from_network_mode(compose, cnt):
         aliases_on_container = [service_name]
         if cnt.get("_aliases"):
             aliases_on_container.extend(cnt.get("_aliases"))
-        net_args.append("--network=bridge")
-        mac_address = cnt.get("mac_address")
-        if mac_address:
-            net_args.append(f"--mac-address={mac_address}")
```
Comment (p12tic) [inline podman_compose.py]: I think we lose mac-address here. Is this deliberate? If yes, please split to separate commit with explanation.

File: podman_compose.py [hunk 4]
```diff
@@ -937,29 +935,16 @@ def get_net_args_from_networks(compose, cnt):
...
+
     multiple_nets = cnt.get("networks", {})
     if not multiple_nets:
         if not compose.default_net:
             # The bridge mode in podman is using the `podman` network.
             # It seems weird, but we should keep this behavior to avoid
             # breaking changes.
-            net_args.append("--network=bridge")
-            if mac_address:
-                net_args.append(f"--mac-address={mac_address}")
```
Comment (p12tic) [inline podman_compose.py]: I think we lose mac-address here. Is this deliberate? If yes, please split to separate commit with explanation.

File: podman_compose.py [hunk 5]
```diff
@@ -984,17 +969,17 @@ def get_net_args_from_networks(compose, cnt):
...
         default_net_name = default_network_name_for_project(compose, net_, is_ext)
         net_name = ext_desc.get("name") or net_desc.get("name") or default_net_name
 
         ipv4 = net_config_.get("ipv4_address")
         ipv6 = net_config_.get("ipv6_address")
-        # custom extension; not supported by docker-compose v3
+
         mac = net_config_.get("x-podman.mac_address")
-        aliases_on_net = norm_as_list(net_config_.get("aliases", []))
+        aliases_on_net = net_config_.get("aliases")
```
Comment (p12tic) [inline podman_compose.py:982]: Probably makes sense to move to separate commit, as it's yet another cleanup.

**Comment by p12tic** [comment]
I'm looking into https://github.com/containers/podman-compose/pull/1087/commits/22454fcf7c1b28b9030c5cff2ff1121ec555688f to review functionality changes.

File: podman_compose.py (hunk 4 above)
Comment (lisongmin) [reply podman_compose.py]: I just missing the mac_address here.

The top-level mac_address should be part of networks, and docker >= 25.0 will ignore this attribute
in favor of the network-scoped mac_address. see [compose spec](https://github.com/compose-spec/compose-spec/blob/main/05-services.md#mac_address-1) for more detail.
    
I'll set the mac_address in the bridge mode here for backward compatibility.

File: podman_compose.py (hunk 2 above)
Comment (lisongmin) [reply podman_compose.py:858]: When using bridge mode, podman-compose will use the default `podman` network.
We do not need to create this network, so just return as other network_mode.

File: podman_compose.py (hunk 5 above)
Comment (lisongmin) [reply podman_compose.py:982]: Just rollback changes here.

**Comment by lisongmin** [comment]
Done. 

Sorry for the poor PR, thanks for your patience in checking the codes. I learn a lot from it.
